discovery-provider/src/api/v1/users.py

```diff
@@ -1425,7 +1425,10 @@ def get(self):
             ns.abort(400, "JWT payload could not be decoded.")
 
         wallet_user_id = get_user_with_wallet(wallet)
-        if not wallet_user_id or wallet_user_id != payload["userId"]:
+        if not wallet_user_id or (
+            wallet_user_id != payload["userId"]
+            and wallet_user_id != decode_string_id(payload["userId"])
```

Comment:
I think this is fine since jons.loads is used above, so if payload["userId"] was a string, it'll remain a string and if it's a number it'll remain a number. But if it's always a number, decode_string_id expects a string, and if it's always a string, comparing it directly to wallet_user_id will always be false.